rpadmin/api_migration.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package rpadmin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+)
+
+const (
+	baseMigrationEndpoint = "/v1/migrations/"
+)
+
+// AddMigration adds a migration to the cluster. It accepts one of InboundMigration or OutboundMigration.
+func (a *AdminAPI) AddMigration(ctx context.Context, migration any) (int, error) {
+	migrationType := reflect.TypeOf(migration)
+	if migrationType != reflect.TypeOf(InboundMigration{}) && migrationType != reflect.TypeOf(OutboundMigration{}) {
+		return 0, fmt.Errorf("invalid migration type: must be either InboundMigration or OutboundMigration")
+	}
+
+	var response AddMigrationResponse
+	if err := a.sendOne(ctx, http.MethodPut, baseMigrationEndpoint, migration, &response, false); err != nil {
+		return -1, err
+	}
+	return response.ID, nil
```

Comment:
Why return the `response.ID` and not the `AddMigrationResponse`? it will be better if we return the struct if more information gets added to the response in the future.